src/java.base/share/classes/java/net/Socket.java

```diff
@@ -650,12 +658,17 @@ public void connect(SocketAddress endpoint) throws IOException {
      *        cause it to wakeup and close the socket. This method will then throw
      *        {@code SocketException} with the interrupt status set.
      * </ol>
+     * <p> Besides above noted cases, the socket will also be closed when a
+     * connection attempt fails with an {@link IOException}.
```

Comment:
Is this a left-over, I assume it should be removed.

Reply:
Doh! :facepalm: Fixed in 424fc8ccd3a495a49f859f0cfc563cf463b3efeb.